src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+                       else
+                         ""
+          )
+          (isJust model.oracle)
+        ]
+        ( let
+
+            makeItem n item =
+              let
+                transitionDuration =
+                  constant.transitionDurationSeconds *
+                  constant.transitionDurationFraction
+
+                offsetTop n =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle ->
+                      oracle.offsetTops
+                      |> List.drop (n-1) -- n is 1-based
+                      |> List.head
+                      |> fromJust "Menu.view: offsetTop"
+
+                offsetHeight n =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle ->
+                      oracle.offsetHeights
+                      |> List.drop (n-1) -- n is 1-based
+                      |> List.head
+                      |> fromJust "Menu.view offsetHeight"
+
+                height =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle -> oracle.menu.bounds.height
+
+                itemDelay =
+
+                  if model.alignment == topLeft ||
+                     model.alignment == topRight then
+
+                      (height - offsetTop n - offsetHeight n) / height * transitionDuration
+                      |> toString
+                      |> flip (++) "s"
+
+                    else
+
+                      ((offsetTop n) / height * transitionDuration)
+                      |> toString
+                      |> (flip (++) "s")
+              in
+
+                    styled li
+                    ( (if item.enabled then
+                          Style.attribute (onClick addr Oracle.decode' (Close' n))
+                        else
+                          Style.attribute (Html.attribute "disabled" "disabled"))
+                      :: cs "mdl-menu__item"
+                      :: css' "transition-delay" itemDelay (isJust model.open)
+                      :: cs' "mdl-js-ripple-effect" model.ripple
+                      :: cs' "mdl-menu__item--full-bleed-divider" item.divider
+                      :: Style.attribute (Html.property "tabindex" (string "-1"))
+                      :: Style.attribute (Ripple.downOn "mousedown" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.downOn "touchstart" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "mouseup" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "mouseleave" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "touchend" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "blur" (Signal.forwardTo addr (Ripple n)))
+                      :: []
```

Comment:
Use the usual `[  , , , , ]` style when you don't need to cons onto something (like here).